OWML.ModHelper.Menus/ModComboInput.cs

```diff
@@ -0,0 +1,149 @@
+﻿using OWML.Common.Menus;
+using OWML.ModHelper.Events;
+using UnityEngine;
+using UnityEngine.UI;
+using System;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModComboInput : ModInput<string>, IModComboInput
+    {
+        private const int FontSize = 36;
+        private const float ScaleDown = 0.75f;
+        private const string XboxPrefix = "xbox_";
+        private static readonly Vector2 normalPivot = new Vector2(0.5f, 0.5f);
+
+        public IModLayoutButton Button { get; }
+        protected readonly IModInputMenu InputMenu;
+        protected readonly TwoButtonToggleElement ToggleElement;
+
+        private string _value;
+        private Vector3 _scale;
+        private UIStyleManager _styleManager;
+        private HorizontalLayoutGroup _layoutGroup;
+
+        public ModComboInput(TwoButtonToggleElement element, IModMenu menu, IModInputMenu inputMenu) : base(element, menu)
+        {
+            ToggleElement = element;
+            InputMenu = inputMenu;
+            _scale = element.GetValue<Button>("_buttonTrue").transform.localScale;
+            Button = new ModLayoutButton(element.GetValue<Button>("_buttonTrue"), menu);
+            Button.OnClick += Open;
+            var noButton = ToggleElement.GetValue<Button>("_buttonFalse");
+            noButton.transform.parent.gameObject.SetActive(false);
+            _layoutGroup = Button.LayoutGroup;
+            ((RectTransform)_layoutGroup.transform).sizeDelta = new Vector2(((RectTransform)Button.Button.transform.parent).sizeDelta.x * 2, ((RectTransform)Button.Button.transform.parent).sizeDelta.y);
+            _styleManager = MonoBehaviour.FindObjectOfType<UIStyleManager>();
+
+            var layoutGroup = Button.Button.transform.parent.parent.GetComponent<HorizontalLayoutGroup>();
+            layoutGroup.childControlWidth = true;
+            layoutGroup.childForceExpandWidth = true;
+            Button.Button.transform.parent.GetComponent<LayoutElement>().preferredWidth = 100;
+        }
+
+        private void UpdateLayout(string currentCombination)
```

Comment:
This method is a bit complicated to read. I think it would be nice if you extract the contents into smaller methods with intuitive names.

Reply:
separated icon adding script

Reply:
probably should generalize and move part of it (as well as AddText) to ModLayoutButton

Reply:
done now